### ISSUE TITLE
Uncaught Exception: $var should not be null for ClassName::method

### ISSUE THREAD

**Issue by sanmai** [vimeo/psalm#5199]
As seen with Psalm 4.5.1:

```
$ php psalm.phar --no-cache --use-baseline=example --update-baseline
Scanning files...
Analyzing files...

PHP Notice:  Undefined offset: 0 in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 523
Notice: Undefined offset: 0 in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 523
PHP Notice:  Trying to get property 'default' of non-object in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 549
Notice: Trying to get property 'default' of non-object in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 549
PHP Notice:  Undefined offset: 1 in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 523
Notice: Undefined offset: 1 in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 523
PHP Notice:  Trying to get property 'default' of non-object in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 549
Notice: Trying to get property 'default' of non-object in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 549
PHP Notice:  Undefined offset: 0 in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 523
Notice: Undefined offset: 0 in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 523
PHP Notice:  Trying to get property 'default' of non-object in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 549
Notice: Trying to get property 'default' of non-object in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 549
PHP Notice:  Undefined offset: 1 in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 523
Notice: Undefined offset: 1 in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 523
PHP Notice:  Trying to get property 'default' of non-object in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 549
Notice: Trying to get property 'default' of non-object in phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FunctionLikeAnalyzer.php on line 549
Uncaught Exception: $storage should not be null for _CLASS_NAME_::run
Stack trace in the forked worker:
#0 phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/ClassAnalyzer.php(771): Psalm\Internal\Codebase\Methods->getStorage()
#1 phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/ClassAnalyzer.php(761): Psalm\Internal\Analyzer\ClassAnalyzer::analyzeClassMethodReturnType()
#2 phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/ClassAnalyzer.php(213): Psalm\Internal\Analyzer\ClassAnalyzer->analyzeClassMethod()
#3 phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/FileAnalyzer.php(170): Psalm\Internal\Analyzer\ClassAnalyzer->analyze()
#4 phar:///.../psalm.phar/src/Psalm/Internal/Codebase/Analyzer.php(267): Psalm\Internal\Analyzer\FileAnalyzer->analyze()
#5 phar:///.../psalm.phar/src/Psalm/Internal/Fork/Pool.php(150): Psalm\Internal\Codebase\Analyzer->Psalm\Internal\Codebase\{closure}()
#6 phar:///.../psalm.phar/src/Psalm/Internal/Codebase/Analyzer.php(314): Psalm\Internal\Fork\Pool->__construct()
#7 phar:///.../psalm.phar/src/Psalm/Internal/Codebase/Analyzer.php(222): Psalm\Internal\Codebase\Analyzer->doAnalysis()
#8 phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(447): Psalm\Internal\Codebase\Analyzer->analyzeFiles()
#9 phar:///.../psalm.phar/src/psalm.php(469): Psalm\Internal\Analyzer\ProjectAnalyzer->check()
#10 phar:///.../psalm.phar/psalm(5): require_once('phar:///home/le...')
#11 /.../psalm.phar(14): require('phar:///home/le...')
#12 {main} in phar:///.../psalm.phar/src/Psalm/Internal/Fork/Pool.php:283
Stack trace:
#0 phar:///.../psalm.phar/src/Psalm/Internal/Fork/Pool.php(311): Psalm\Internal\Fork\Pool->readResultsFromChildren()
#1 phar:///.../psalm.phar/src/Psalm/Internal/Codebase/Analyzer.php(349): Psalm\Internal\Fork\Pool->wait()
#2 phar:///.../psalm.phar/src/Psalm/Internal/Codebase/Analyzer.php(222): Psalm\Internal\Codebase\Analyzer->doAnalysis()
#3 phar:///.../psalm.phar/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(447): Psalm\Internal\Codebase\Analyzer->analyzeFiles()
#4 phar:///.../psalm.phar/src/psalm.php(469): Psalm\Internal\Analyzer\ProjectAnalyzer->check()
#5 phar:///.../psalm.phar/psalm(5): require_once('phar:///home/le...')
#6 /.../psalm.phar(14): require('phar:///home/le...')
#7 {main}
(Psalm 4.x-dev@ crashed due to an uncaught Throwable)
```

```xml
<?xml version="1.0"?>
<psalm
    errorLevel="6"
    resolveFromConfigFile="true"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns="https://getpsalm.org/schema/config"
    xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
>
    <projectFiles>
        <directory name="app" />
        <directory name="lib" />
        <directory name="tests" />
        <!-- and few more -->
        <ignoreFiles>
            <directory name="vendor" />
        </ignoreFiles>
    </projectFiles>
</psalm>
```
And:
```
$ php -v
PHP 7.4.14 (cli) (built: Jan 12 2021 10:26:28) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.14, Copyright (c), by Zend Technologies
    with blackfire v1.49.0~linux-x64-non_zts74, https://blackfire.io, by Blackfire
```

**Comment by weirdan** [comment]
Is that a literal `_CLASS_NAME_`, or did you use that to obfuscate the actual class name?

**Comment by sanmai** [comment]
This is a literal `_CLASS_NAME_`

> Uncaught Exception: $storage should not be null for _CLASS_NAME_::execute (75%)

**Comment by weirdan** [comment]
> I the reasons for the exception are unclear here, I will try to make a reproducer.

Please do.

**Comment by Chekote** [comment]
I think [this](https://github.com/vimeo/psalm/issues/4836#issuecomment-792177157) is fixed by 817d1cf.

**Comment by orklah** [comment]
> I think this is fixed by [817d1cf](https://github.com/vimeo/psalm/commit/817d1cf3fe9f3e3141bcb7f0a5734d2714ecdf78).

Most probably! Thanks!
